core/templates/dev/head/editor/RouterServices.js

```diff
@@ -137,34 +155,34 @@ oppia.factory('routerService', [
       },
       navigateToMainTab: function(stateName) {
         _savePendingChanges();
-
         if (_getCurrentStateFromLocationPath() === stateName) {
           return;
         }
 
-        var _actuallyNavigate = function(newStateName) {
-          if (newStateName) {
-            editorContextService.setActiveStateName(newStateName);
-          }
-          $location.path('/gui/' + editorContextService.getActiveStateName());
-          $window.scrollTo(0, 0);
-        };
-
         if (_tabs.active === MAIN_TAB) {
           $('.oppia-editor-cards-container').fadeOut(function() {
-            _actuallyNavigate(stateName);
+            _actuallyNavigate('gui', stateName);
             $rootScope.$apply();
             $timeout(function() {
               $('.oppia-editor-cards-container').fadeIn();
             }, 150);
           });
         } else {
-          _actuallyNavigate(stateName);
+          _actuallyNavigate('gui', stateName);
         }
       },
-      navigateToPreviewTab: function() {
-        _savePendingChanges();
-        $location.path('/preview');
+      navigateToPreviewTab: function(stateName) {
+        if (_tabs.active === PREVIEW_TAB) {
```

Comment:
Ah, no -- for the main tab, that code is there to handle changes between states in the exploration, since that's a common action on that tab. For the preview tab, this isn't really the case, so it's fine to remove it from here, but I would leave the main tab stuff as it is for now. Thanks for asking!